### ISSUE TITLE
Mockito fails to verify Kotlin function types on Robolectric 4.3

### ISSUE THREAD

**Issue by jameswald** [robolectric/robolectric#5076]
### Description

Kotlin function types can be mocked and verified in JVM unit tests and on Android devices, but these tests fail on the Robolectric environment. Mockito claims that `invoke()` is not being called after `verify()`.

### Steps to Reproduce

Run the following test case:

```
package com.example.test

import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify

@RunWith(AndroidJUnit4::class)
class Issue5076Test {

    @Suppress("UNCHECKED_CAST")
    private val function = mock(Function1::class.java) as (String) -> Unit

    @Test
    fun test1() {
        function.invoke("test")
        verify(function).invoke("test")
    }

    @Test
    fun test2() {
        function.invoke("test")
        verify(function).invoke("foo")
    }
}
```

Test 1 passes and test 2 fails:

```
org.mockito.exceptions.misusing.UnfinishedVerificationException: 
Missing method call for verify(mock) here:
-> at com.example.test.Issue5076Test.test1(Issue5076Test.kt:18)

Example of correct verification:
    verify(mock).doSomething()

Also, this error might show up because you verify either of: final/private/equals()/hashCode() methods.
Those methods *cannot* be stubbed/verified.
Mocking methods declared on non-public parent classes is not supported.
```

Both test 1 and test 2 pass when they are run individually, however, test 2 should fail because the argument does not match ("test" != "foo"). The tests behave correctly when `AndroidJUnit4::class` is replaced with `JUnit4::class`:

```
Argument(s) are different! Wanted:
function1.invoke("foo");
-> at com.example.test.Issue5076Test.test2(Issue5076Test.kt:24)
Actual invocation has different arguments:
function1.invoke("test");
-> at com.example.test.Issue5076Test.test2(Issue5076Test.kt:23)
    at com.example.test.Issue5076Test.test2(Issue5076Test.kt:24)
```

### Robolectric & Android Version

Robolectric 4.3
Android 28
Mockito 2.25.0 (org.mockito:mockito-inline)

### Link to a public git repo demonstrating the problem:

See test case above.

**Comment by XIII-th** [comment]
While this issue is open, you can define an interface that implements a lambda and use it instead
```kotlin
/**
 * This interface only for testing
 */
private interface TestInterface: (String) -> Unit

@RunWith(AndroidJUnit4::class)
class Issue5076Test {

    @Suppress("UNCHECKED_CAST")
    private val function = mock(TestInterface::class.java) as (String) -> Unit
    /* or 
    private val function = mock(TestInterface::class.java)
     */

    @Test
    fun test1() {
        function.invoke("test")
        verify(function).invoke("test")
    }

    @Test
    fun test2() {
        function.invoke("test")
        verify(function).invoke("foo")
    }
}
```

**Comment by hoisie** [comment]
Is this still an issue? mockito-inline 3.5.7 should work well with Robolectric now.

**Comment by XIII-th** [comment]
@hoisie yes. I've just checked it with Robolectric 4.3.1 and mockito-inline 3.5.7

**Comment by hoisie** [comment]
Did you try with 4.4?

**Comment by jameswald** [comment]
I'm currently working on migrating to Robolectric 4.4 and the latest mockito-inline. I'll close out this issue if those updates resolve this issue.

**Comment by hoisie** [comment]
If this is still an issue, could someone make a PR to Robolectric that adds a new project `integration_tests/mockito-kotlin` and add that failing test using `@Ignore`? Also update mockito to 3.5.11 as it contains a Robolectric related fix. The mockito devs are super helpful and responsive to these types of issues.

**Comment by ZSmallX** [comment]
> If this is still an issue, could someone make a PR to Robolectric that adds a new project `integration_tests/mockito-kotlin` and add that failing test using `@Ignore`? Also update mockito to 3.5.11 as it contains a Robolectric related fix. The mockito devs are super helpful and responsive to these types of issues.

I added a PR https://github.com/robolectric/robolectric/pull/7086 to add tests to verify this question.

This issue can be solved by upgrading software infra ( can be seen in PR). 

@jameswald Could you please help checking if it's still an issue?

**Comment by jameswald** [comment]
I don't have access to the original project for testing but I recall the updates solving the reported issue. The tests you added appear to cover this scenario so this issue can be closed by that. Thanks for writing those!

**Comment by jameswald** [comment]
Closes #5076
